library/Control/Monad/AWS/Class.hs

```diff
@@ -21,13 +23,29 @@ import Data.Typeable (Typeable)
 -- - "Control.Monad.AWS.ViaReader"
 -- - "Control.Monad.AWS.ViaMock"
 class Monad m => MonadAWS m where
+  -- |
+  --
+  -- @since 0.1.0.0
   sendEither
     :: (AWSRequest a, Typeable a, Typeable (AWSResponse a))
     => a
     -> m (Either Error (AWSResponse a))
 
+  -- |
+  --
+  -- @since 0.1.0.0
   awaitEither
     :: (AWSRequest a, Typeable a)
     => Waiter.Wait a
     -> a
     -> m (Either Error Waiter.Accept)
+
+  -- |
+  --
+  -- @since 0.1.1.0
+  withAuth :: (AuthEnv -> m a) -> m a
+
+  -- |
+  --
+  -- @since 0.1.1.0
+  modified :: (Env -> Env) -> m a -> m a
```

Comment:
I'd be inclined to include "env" in the name, maybe `localEnv`? I don't quite follow your comment about local not always being the case.

Reply:
> I'd be inclined to include "env" in the name

- `ReaderAWS` has `modified = local`
- `MockAWS` does not

Hence, it's "not always the case" that `modified` is just there to be `local`. In `MockAWS`, `modified` does nothing because there is nothing to modify.

Reply:
`modifiedEnv` SGTM.